useragent/cicd.go

```diff
@@ -0,0 +1,72 @@
+package useragent
+
+import (
+	"os"
+	"sync"
+)
+
+type envVar struct {
+	name          string
+	expectedValue string
```

Comment:
Nit: can we just comment here that if expectedValue is empty string, we only check whether the environment variable is defined, and if nonempty, we check that the value is equal to the provided value?